cfme/tests/automate/test_instance.py

```diff
@@ -354,3 +354,117 @@ def test_quota_source_value(request, entity, search, copy_quota_instance, generi
     provision_request.wait_for_request(method='ui')
     request.addfinalizer(lambda: provision_request.remove_request(method="rest"))
     assert result.validate(wait="60s")
+
+
+@pytest.mark.tier(1)
+@pytest.mark.meta(automates=[1441353])
+@pytest.mark.parametrize("process", ["miq_stop", "miq_abort"])
+def test_miq_stop_abort_with_state_machines(request, process, domain, klass, namespace):
+    """
+    Bugzilla:
+        1441353
+
+    Polarion:
+        assignee: ghubale
+        initialEstimate: 1/8h
+        caseimportance: high
+        caseposneg: positive
+        testtype: functional
+        startsin: 5.9
+        casecomponent: Automate
+        tags: automate
+    """
+    state1 = fauxfactory.gen_alpha()
+    state2 = fauxfactory.gen_alpha()
+    state3 = fauxfactory.gen_alpha()
+    for state in [state1, state2, state3]:
+        klass.schema.add_fields({'name': state, 'type': 'State'})
+
+    # Adding method to execute via parent instance(state machine instance)
+    method = klass.methods.create(
+        name=fauxfactory.gen_alphanumeric(),
+        display_name=fauxfactory.gen_alphanumeric(),
+        location='inline',
+        script="""
+               \n$evm.log(:info, "Hello from method of parent instance")
+                \nexit MIQ_STOP
+                """
+    )
+    request.addfinalizer(method.delete_if_exists)
+
+    # Adding parent instance to call automate method and child instances in other class
+    instance = klass.instances.create(
+        name=fauxfactory.gen_alphanumeric(),
+        display_name=fauxfactory.gen_alphanumeric(),
+        description=fauxfactory.gen_alphanumeric(),
+        fields={state3: {"value": f"METHOD::{method.name}"}}
+    )
+    request.addfinalizer(instance.delete_if_exists)
+
+    # Creating other class for two child instances
+    klass2 = namespace.classes.create(
+        name=fauxfactory.gen_alpha(),
+        display_name=fauxfactory.gen_alpha(),
+        description=fauxfactory.gen_alpha()
+    )
+    request.addfinalizer(klass2.delete_if_exists)
+
+    for state in [state1, state2]:
+        klass2.schema.add_fields({'name': state, 'type': 'State'})
+
+    # Creating three child methods to execute via child instances
+    child_method = list(map(lambda num: klass2.methods.create(
```

Comment:
I saw that for the second parametrized test, new `state1`, `state2` and `state3` are created inside the same `klass` that was used for the first test..
So if everything until creating this child method is common for both the parameters, why not just use the same parent klass/method/instance for both the parametrized test?

Reply:
Thanks @valaparthvi for this suggestion. 
I am making changes as per this suggestion.